src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -499,26 +476,63 @@ private void handleCloudWatchAttemptStatus(CloudWatchAttempt cloudWatchAttempt)
         isCurrentlyUploading.set(false);
     }
 
+    private void deleteFilesInComponents(Map<String, Set<String>> fileDeleteSet) {
+        for (Map.Entry<String, Set<String>> entry : fileDeleteSet.entrySet()) {
+            String key = entry.getKey();
+            Set<String> fileNames = entry.getValue();
+            fileNames.stream().map(LogFile::new).forEach(file -> {
+                if (!lastComponentUploadedLogFileInstantMap.containsKey(key)
+                        || lastComponentUploadedLogFileInstantMap.get(key)
+                        .isBefore(Instant.ofEpochMilli(file.lastModified()))) {
+                    lastComponentUploadedLogFileInstantMap.put(key, Instant.ofEpochMilli(file.lastModified()));
+                }
+                // delete the entry if one file is read and uploaded completely
+                fileHashNamePairs.remove(file.hashString());
+            });
+            if (!componentLogConfigurations.containsKey(key)) {
+                continue;
+            }
+            ComponentLogConfiguration componentLogConfiguration = componentLogConfigurations.get(key);
+            if (!componentLogConfiguration.isDeleteLogFileAfterCloudUpload()) {
+                continue;
+            }
+            fileNames.forEach(fileName -> {
+                try {
+                    boolean successfullyDeleted = Files.deleteIfExists(Paths.get(fileName));
+                    if (successfullyDeleted) {
+                        logger.atDebug().log("Successfully deleted file with name {}", fileName);
+                    } else {
+                        logger.atWarn().log("Unable to delete file with name {}", fileName);
+                    }
+                } catch (IOException e) {
+                    logger.atError().cause(e).log("Unable to delete file with name: {}", fileName);
+                }
+            });
+        }
+    }
+
     private void processCloudWatchAttemptLogInformation(Map<String, Set<String>> completedLogFilePerComponent,
                                                         Map<String, CurrentProcessingFileInformation>
                                                                 currentProcessingLogFilePerComponent,
                                                         CloudWatchAttemptLogInformation attemptLogInformation,
-                                                        String fileName,
+                                                        String fileHash,
                                                         CloudWatchAttemptLogFileInformation
                                                                 cloudWatchAttemptLogFileInformation) {
-        File file = new File(fileName);
+        String fileName = fileHashNamePairs.get(fileHash);
```

Comment:
I thought one benefit of using the hash was so that we did not rely anymore on fileNames... It might happen as we saw that the fileNames might not change but the hash might (depending on the rotation strategy). So we might get a scenario where we have multiple hashes pointing to the same file